www/md_base/src/app/common/directives/forcebuildform/forcebuildform.directive.coffee

```diff
@@ -0,0 +1,158 @@
+class ForcebuildForm extends Directive
+
+    constructor: ->
+        return {
+            restrict: 'E'
+            controller: '_ForcebuildFormController'
+            controllerAs: 'form'
+            bindToController: true
+            scope:
+                model: '='
+                fields: '='
+                errors: '='
+        }
+
+class _ForcebuildForm extends Controller
+
+    constructor: ($scope, $element, $compile) ->
+        elem = $compile(@buildfields())($scope)
+        $element.append elem
+
+    element: (tag, attributes) ->
+        elem = angular.element("<#{tag}></#{tag}>")
+        elem.attr(attributes) if attributes
+        return elem
+
+    labelElement: (text) ->
+        text = text.trim().replace /:$/, ''
+        return null if not text
+        label = @element 'label'
+        label.text(text)
+        return label
+
+    buildfields: ->
+        elem = @element 'div', class: 'inner'
+        for field in @fields
+            elem.append @buildfield(field)
+        return elem
+
+    buildfield: (field) ->
+        if field.type == 'nested'
+            # Build layout
+            switch field.layout
+                when 'vertical'
+                    return @verticalLayout(field)
+                when 'tab'
+                    return @tabLayout(field)
+                else
+                    return @simpleLayout(field)
+        else
+            # Build normal fields
+            switch field.type
+                when 'textarea'
+                    return @textareaField(field)
+                when 'int'
+                    return @intField(field)
+                when 'list'
+                    return @listField(field)
+                when 'bool'
+                    return @boolField(field)
+                else
+                    return @textField(field)
```

Comment:
This method is obviously much more simple than the way I used previously.
It however does not address the requirement that people can define custom field for their own need.
See for example the CodeParameter instance.
The idea is that the form is filled with custom directives whose name is "#{field.type}Field". So my plugin just has to register a new directive in order to create a new field.


I think you can achieve this while keeping the simpler plugins like this.
You just have to change the default case.
```
when 'text'
    return @textField(field)
else
    return @customField(field)
```
and customField will just generate

```
    customField: (field) ->
          elem = @element field.type + "Field"
          elem.$scope.field = field # I am sure this wont work, but you'll have to find a way to pass field as a parameter to the custom field
          return
```

I'd like you to eventually make it work with codeparameter plugin, without having to modify it.
I realize that this require to work first on making the plugin system work.
maybe that is too much work.